internal/contour/listener.go

```diff
@@ -403,10 +406,12 @@ func (v *listenerVisitor) visit(vertex dag.Vertex) {
 		v.listeners[ENVOY_HTTPS_LISTENER].FilterChains = append(v.listeners[ENVOY_HTTPS_LISTENER].FilterChains,
 			envoy.FilterChainTLS(vh.VirtualHost.Name, downstreamTLS, filters))
 
-		// If this VirtualHost has enabled the fallback certificate then set a default FilterChain which will allow
-		// routes with this vhost to accept non SNI TLS requests
+		// If this VirtualHost has enabled the fallback certificate then set a default
+		// FilterChain which will allow routes with this vhost to accept non-SNI TLS requests.
+		// Note that we don't add the misdirected requests filter on this chain because at this
```

Comment:
Presumably this means that if you're a client old enough not to support SNI, then you're a client that's not going to coalesce requests onto a single TLS session?

Reply:
The misdirected requests filter works by matching the hostname we get against the hostname we expect. In the fallback case, there are a set of hostnames that could be expected (all the vhosts that set the fallback enabled). At this point, however, we don't know  all those hostnames so we can't make the right filter chain.

So I think we could implement this; we would need to update the Lua code to match against a set of expected hostnames, and we would need to delay creating the fallback filter until we have seen all the vhosts (so we know the full set that enabled fallback).